### DISCUSSION TITLE
Helper for url

### DISCUSSION THREAD

**Discussion by battulga0719** [awcodes/filament-curator#494] [Ideas]
Hello. is there a way to get url in curator cause what i want is for example if i have a div with a background image i want to attach curator link to that background url('') without using x-curator-curation or x-curator-glide i will just use a Curator::generateUrl($imageId, ['options']) so i can just call it to get a thumbnail url or banner image.

**Comment by awcodes** [comment]
https://github.com/awcodes/filament-curator/blob/3.x/src/Glide/GlideBuilder.php

**Reply by battulga0719** [reply]
wow. thank you very much 👍

**Reply by battulga0719** [reply]
@awcodes is there a recommended way to use it? or should i just call it can use it.

**Reply by awcodes** [reply]
This is how I'm doing it in one of my projects. Not the prettiest. lol.

```blade
@if (filled($backgroundImage))
    style="background-image: url('{{ \Awcodes\Curator\is_media_resizable($backgroundImage['path']) ? GlideBuilder::make()->toUrl($backgroundImage['path'] ?? '') : asset($backgroundImage['path']) }}')"
@endif
```

**Reply by awcodes** [reply]
and with a regular img tag:
```php
<img
    src="{{
        \Awcodes\Curator\is_media_resizable($image['path'])
            ? GlideBuilder::make()
                ->format('webp')
                ->width(640)
                ->toUrl($image['path'] ?? '')
            : asset($image['path'])
    }}"
    alt="{{ $image['alt'] }}"
    width="{{ $image['width'] }}"
    height="{{ $image['height'] }}"
/>

**Reply by battulga0719** [reply]
oh that's cool thank you sir 💯 
as long as it works it's not stupid

**Reply by battulga0719** [reply]
Hello. good day sir. i just updated and tested the component and 1 question occurred to me how do you make $image because curator stores id and if i have a 20image on single page i'll have to connect to db 20times or i have to use a caching. is there a other way to get path?

**Reply by awcodes** [reply]
Sounds like you need a relationship for the images and the page. Then you can eager load them all.

**Reply by battulga0719** [reply]
some images are in a settings page and other shinanigans like in a repeater so i cant rely on a relationship so i need more reliable way to cache and get from it current im doing it like this
<div class="chocolat-parent">
        @php($media = cache()->remember('curator-'.$item->image, now()->addMonth(4), function()use($item){
            return app(config('curator.model'))::find($item->image);
        }))
        <a class="chocolat-image relative flex" href="{{ $media->url ?? null }}" rel="nofollow">
            <x-curator-glider
                class="w-full object-cover object-center aspect-[16/7] rounded-tr-2xl rounded-bl-2xl"
                :media="$item->image"
                width="1024"
                fallback="default"
            />
        </a>
    </div>
    it looks terrible but it keeps the job done for now atleast

**Comment by battulga0719** [comment]
i guess this my skill issue. thank you sir for quick reply and fix
